static/src/javascripts/projects/common/modules/ui/bannerPicker.spec.js

```diff
@@ -0,0 +1,108 @@
+// @flow
+import { init, _ } from 'common/modules/ui/bannerPicker';
+
+jest.mock('ophan/ng', () => {});
+
+const shortTrueCheck = () =>
+    new Promise(resolve => {
+        setTimeout(() => {
+            resolve(true);
+        }, 100);
+    });
+const shortFalseCheck = () =>
+    new Promise(resolve => {
+        setTimeout(() => {
+            resolve(false);
+        }, 100);
+    });
+const longFalseCheck = () =>
+    new Promise(resolve => {
+        setTimeout(() => {
+            resolve(false);
+        }, 200);
+    });
+const longTrueCheck = () =>
+    new Promise(resolve => {
+        setTimeout(() => {
+            resolve(true);
+        }, 200);
+    });
+const extraLongTrueCheck = () =>
+    new Promise(resolve => {
+        setTimeout(() => {
+            resolve(true);
+        }, 2500);
+    });
+const extraLongFalseCheck = () =>
+    new Promise(resolve => {
+        setTimeout(() => {
+            resolve(false);
+        }, 2500);
+    });
```

Comment:
I'd be hesitant to add tests that could take several seconds to execute (if that's what is happening here!). Is there a way to rewrite using Jest's [`advanceTimersByTime`](https://facebook.github.io/jest/docs/en/timer-mocks.html#advance-timers-by-time) method?

Reply:
A fair point, I'll investigate timer methods offered by Jest.